ScoreSaber/Core/Services/LeaderboardService.cs

```diff
@@ -54,29 +65,27 @@ private string GetLeaderboardUrl(BeatmapKey beatmapKey, PlatformLeaderboardsMode
-                    case PlatformLeaderboardsModel.ScoresScope.Global:
-                        url = $"{url}/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
-                        break;
-                    case PlatformLeaderboardsModel.ScoresScope.AroundPlayer:
-                        url = $"{url}/around-player/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}";
-                        hasPage = false;
-                        break;
-                    case PlatformLeaderboardsModel.ScoresScope.Friends:
-                        url = $"{url}/around-friends/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
-                        break;
-                }
-            } else {
-                if(Plugin.Settings.locationFilterMode.ToLower() == "region") {
-                    url = $"{url}/around-region/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
-                }
-                else if(Plugin.Settings.locationFilterMode.ToLower() == "country") {
-                    url = $"{url}/around-country/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
-                } else {
-                    Plugin.Log.Error("Invalid location filter mode, falling back to country");
-                    url = $"{url}/around-country/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
-                }
+            switch (scope) {
+                case ScoreSaberLeaderboardViewController.ScoreSaberScoresScope.Global:
+                    url = $"{url}/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
+                    break;
+                case ScoreSaberLeaderboardViewController.ScoreSaberScoresScope.Player:
+                    url = $"{url}/around-player/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}";
+                    hasPage = false;
+                    break;
+                case ScoreSaberLeaderboardViewController.ScoreSaberScoresScope.Friends:
+                    url = $"{url}/around-friends/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
+                    break;
+                case ScoreSaberLeaderboardViewController.ScoreSaberScoresScope.Area:
+                    if (Plugin.Settings.locationFilterMode.ToLower() == "region") {
+                        url = $"{url}/around-region/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
+                    } else if (Plugin.Settings.locationFilterMode.ToLower() == "country") {
+                        url = $"{url}/around-country/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
+                    } else {
+                        Plugin.Log.Error("Invalid location filter mode, falling back to country");
+                        url = $"{url}/around-country/{leaderboardId}/mode/{gameMode}/difficulty/{difficulty}?page={page}";
+                    }
+                    break;
```

Comment:
thats there just in case someone edits the settings file to an invalid location. We could add the region as another button but it might be better to do after the pr imo.